### DISCUSSION TITLE
Change x-axis in ranking plot

### DISCUSSION THREAD

**Discussion by Tomoya-Iizawa** [scikit-hep/cabinetry#463] [Q&A]
Hi,
I am creating ranking plot using cabinetry.visualize.ranking, is there any way/option to set the range of x-axis by hand?
Best,
Tomoya

**Comment by alexander-held** [comment]
Hi, the function returns the figure, from which you should be able to access the axis. Something along the lines of https://github.com/scikit-hep/cabinetry/discussions/457#discussioncomment-8234035 should also work here. I cannot give it a try myself at the moment, please let me know if this already works or I will have a closer look.

**Reply by Tomoya-Iizawa** [reply]
OK, I have expected that kind of implementaion would be necessary :). Thank you for your quick response!

**Reply by alexander-held** [reply]
Out of curiosity: the range is automatically determined based on the different impacts observed, is there a specific reason you are looking to change it / is the automatic version not working well?

**Reply by Tomoya-Iizawa** [reply]
![outRange](https://github.com/scikit-hep/cabinetry/assets/61150311/eea28f58-088e-43ef-9ac5-8a204e9e5feb)
As an attached figure, I see the case some of them are in the out of range. maybe normalization factors are not used for ranging?

**Reply by alexander-held** [reply]
You are right, that looks like a bug: the code currently looks only at pre-fit impact to evaluate the range, and normalization factors do not have that defined. Should be easy to solve, I will get this fixed. Thanks for spotting!

**Reply by alexander-held** [reply]
#465 will fix this.

**Reply by alexander-held** [reply]
Please give the updated version a try and let me know in case of any remaining issues!

**Reply by Tomoya-Iizawa** [reply]
thanks! I am using the version with custom modified function discussed [here](https://github.com/scikit-hep/cabinetry/pull/385), does this version include this feature?

**Reply by alexander-held** [reply]
I've just rebased that branch, so now #385 does contain this feature.
